src/abstractdataframe/join.jl

```diff
@@ -272,10 +274,10 @@ join(name, job2, on = :ID => :identifier)
 function Base.join(df1::AbstractDataFrame,
                    df2::AbstractDataFrame;
                    on::Union{<:OnType, AbstractVector{<:OnType}} = Symbol[],
-                   kind::Symbol = :inner)
+                   kind::Symbol = :inner, makeunique::Bool=false)
```

Comment:
Why is this needed? By definition, joins match columns with identical names, so that shouldn't be a problem? Anyway it should be in the docstring.

Reply:
It is needed. Join matches on `on` keyword argument not on columns having identical names. And we can have columns with identical names on which we do not join.
See the example as it works now:
```
julia> x = DataFrame(rand(3,3))
3×3 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │
├─────┼──────────┼──────────┼──────────┤
│ 1   │ 0.97283  │ 0.200961 │ 0.591387 │
│ 2   │ 0.391169 │ 0.532865 │ 0.630848 │
│ 3   │ 0.090204 │ 0.320481 │ 0.775537 │

julia> y = DataFrame(rand(3,3))
3×3 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │
├─────┼──────────┼──────────┼──────────┤
│ 1   │ 0.470765 │ 0.468833 │ 0.199083 │
│ 2   │ 0.303053 │ 0.992384 │ 0.140139 │
│ 3   │ 0.284372 │ 0.816792 │ 0.675941 │

julia> x[:id] = 1:3
1:3

julia> y[:id] = 1:3
1:3

julia> join(x, y, on=:id)
3×7 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │ id │ x1_1     │ x2_1     │ x3_1     │
├─────┼──────────┼──────────┼──────────┼────┼──────────┼──────────┼──────────┤
│ 1   │ 0.97283  │ 0.200961 │ 0.591387 │ 1  │ 0.470765 │ 0.468833 │ 0.199083 │
│ 2   │ 0.391169 │ 0.532865 │ 0.630848 │ 2  │ 0.303053 │ 0.992384 │ 0.140139 │
│ 3   │ 0.090204 │ 0.320481 │ 0.775537 │ 3  │ 0.284372 │ 0.816792 │ 0.675941 │
```
I will improve the docstring.